source/grid/grid_tools.cc

```diff
@@ -848,53 +848,111 @@ namespace GridTools
                              std::vector<CellData<dim>> &  cells,
                              SubCellData &                 subcelldata,
                              std::vector<unsigned int> &   considered_vertices,
-                             double                        tol)
+                             const double                  tol)
   {
-    // create a vector of vertex
-    // indices. initialize it to the identity,
-    // later on change that if necessary.
+    AssertIndexRange(2, vertices.size());
+    // create a vector of vertex indices. initialize it to the identity, later
+    // on change that if necessary.
     std::vector<unsigned int> new_vertex_numbers(vertices.size());
-    for (unsigned int i = 0; i < vertices.size(); ++i)
-      new_vertex_numbers[i] = i;
+    std::iota(new_vertex_numbers.begin(), new_vertex_numbers.end(), 0);
 
-    // if the considered_vertices vector is
-    // empty, consider all vertices
+    // if the considered_vertices vector is empty, consider all vertices
     if (considered_vertices.size() == 0)
       considered_vertices = new_vertex_numbers;
-
     Assert(considered_vertices.size() <= vertices.size(), ExcInternalError());
 
+    // The algorithm below improves upon the naive O(n^2) algorithm by first
+    // sorting vertices by their value in one component and then only
+    // comparing vertices for equality which are nearly equal in that
+    // component. For example, if @p vertices form a cube, then we will only
+    // compare points that have the same x coordinate when we try to find
+    // duplicated vertices.
+
+    // Start by finding the longest coordinate direction. This minimizes the
+    // number of points that need to be compared against each-other in a
+    // single set for typical geometries.
+    const BoundingBox<spacedim> bbox(vertices);
```

Comment:
:-)

Reply:
I was looking for `BoundingBox(begin, end)` when I first wrote this patch. `BoundingBox(container)` also works.